accounts-db/src/accounts_db.rs

```diff
@@ -3794,6 +3796,17 @@ impl AccountsDb {
                         // not exist in the re-written slot. Unref it to keep the index consistent with
                         // rewriting the storage entries.
                         unrefed_pubkeys.push(pubkey);
+                        if ref_count == 2 && slot_list.len() == 1 {
+                            // this is the last remaining older entry for a zero lamport account. When this account at this slot is shrunk away, the
+                            // zero lamport account that is still alive can be marked dead, removed from the index, and then shrunk away.
+                            let entry = slot_list.first().unwrap();
+                            if entry.1.is_zero_lamport() {
+                                self.add_uncleaned_pubkeys_after_shrink(
+                                    entry.0,
+                                    [*pubkey].into_iter(),
+                                );
+                            }
```

Comment:
abs runs: flush, clean, shrink, ... This code is running in shrink. It could be a while before we get back to clean. Depending on max clean root and such, it could be quite a while before this gets triggered. I don't think there is a way to keep it in the in mem index anyway.